### PR TITLE
delete all server-to-server shares if a user gets deleted

### LINKED ISSUE #13535: Clean up S2S hooks when remote user is deleted
We need to cleanup S2S hooks when remote users are deleted.

@schiesbn, as discussed.

### PR DESCRIPTION
**Pull request by schiessle**
fix https://github.com/owncloud/core/issues/13535

cc @jnfrmarks

### CODE REVIEW

**Comment by karlitschek** [comment]
:+1:  as discussed

File: apps/files_sharing/lib/external/manager.php [hunk 1]
```diff
@@ -261,6 +261,33 @@ public function removeShare($mountPoint) {
 	}
 
 	/**
+	 * remove all shares for user $uid if the user was deleted
+	 *
+	 * @param type $uid
+	 * @return type
```
Comment (LukasReschke) [inline apps/files_sharing/lib/external/manager.php]: `type`?

File: apps/files_sharing/lib/external/manager.php [hunk 2]
```diff
@@ -261,6 +261,33 @@ public function removeShare($mountPoint) {
...
+	 */
+	public function removeUserShares($uid) {
```
Comment (DeepDiver1975) [inline apps/files_sharing/lib/external/manager.php]: @schiesbn where is the unit test for this? A must have - please add - THX

Comment (schiesbn) [inline apps/files_sharing/lib/external/manager.php]: added

Comment (DeepDiver1975) [inline apps/files_sharing/lib/external/manager.php]: THX a lot!

**Comment by DeepDiver1975** [comment]
unit tests are properly executed on oci

``sh
deepdiver@alien:~/Development/ownCloud/core-autotest [s2s_delete_shares]$ ./autotest.sh oci apps/files_sharing/tests/server2server.php 
Using database oc_autotest
Setup environment for oci testing ...
drop the database

Benutzer wurde gelöscht.

create the database

Benutzer wurde erstellt.


Benutzer wurde geändert.


Benutzerzugriff (Grant) wurde erteilt.

INDEX
END INDEX
Testing with oci ...
PHPUnit 4.3.5 by Sebastian Bergmann.

Configuration read from /home/deepdiver/Development/ownCloud/core-autotest/tests/phpunit-autotest.xml

......

Time: 25.34 seconds, Memory: 39.00Mb

OK (6 tests, 36 assertions)

**Comment by MorrisJobke** [comment]
#13568 was merged

**Pull request closed without merging**